spec/models/book_spec.rb

```diff
@@ -99,6 +99,84 @@
     end
   end
 
+  describe ".count_in_leap_years" do
+    it "Determines how many books were created during a leap year" do
+      book = Book.create!(name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1))
+
+      expect(Book.count_in_leap_years).to eq(1)
+
+      book.update!(created_at: Date.new(1700, 1, 1))
+      # book.update_column(:created_at, Date.new(1700, 1, 1))
+
+      expect(Book.count_in_leap_years).to eq(0)
+
+      book.update!(created_at: Date.new(2016, 1, 1))
+
+      expect(Book.count_in_leap_years).to eq(1)
+
+      book.update!(created_at: Date.new(2111, 1, 1))
+
+      expect(Book.count_in_leap_years).to eq(0)
+    end
+  end
+
+  describe ".count_per_year" do
+    it "Determines how many books were created per year" do
+      expect(Book.count_per_year).to eq({ })
+
+      book = Book.create!(name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1))
+
+      expect(Book.count_per_year).to eq({ 2000 => 1})
+
+      book2 = Book.create!(name: "Freedom", isbn: 1919191919, created_at: Date.new(2021, 1, 1))
+
+      expect(Book.count_per_year).to eq({ 2000 => 1, 2021 => 1})
+
+      # I can reasonably expect there to be either no books, a book, or multiple books.
+      # This test covers all of those options. Is this enough?
+    end
+  end
```

Comment:
I'd probably do dates that were closer together on the borderlines of a year. EG: 31st Dec and 1st Jan.

Those two should be enough.

No need to put them in local variables